compiler/rustc_codegen_rmc/src/codegen/function.rs

```diff
@@ -158,30 +148,49 @@ impl<'tcx> GotocCtx<'tcx> {
             ),
         };
 
-        // Now that we have the tuple, write the individual component locals
-        // back to it as a GotoC struct.
+        // When we codegen the function signature elsewhere, we will codegen the untupled version.
+        // We then marshall the arguments into a local variable holding the expected tuple.
```

Comment:
nit: I personally don't associate the work "marshall" with what's happening here - is this a common definition?

Reply:
https://docs.microsoft.com/en-us/cpp/dotnet/overview-of-marshaling-in-cpp?view=msvc-170
It is a term used to indicate translating data between one calling convention and another.